tests/glsl-intrinsic/shader-invocation-control/shader-invocation-control.slang

```diff
@@ -0,0 +1,40 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv -emit-spirv-directly
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+
+#version 460
+
+//TEST_INPUT:ubuffer(data=[2], stride=4):out,name=raceCondition
+buffer MyBlockName1
+{
+    uint data[];
+} raceCondition;
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 4) in;
+
+bool testBarrier()
+{
+    raceCondition.data[0] = 3;
+    barrier();
```

Comment:
I thought "barrier()" is for "groupshared" variables not the buffers.
Is it not the case?

Reply:
you are correct, my current implementation for barrier is wrong
(should be OpControlBarrier Workgroup Workgroup AcquireRelease|WorkgroupMemory